README.md

```diff
@@ -21,16 +21,60 @@ The Gateway Utility Board Rev. 2 (GUB2) is responsible for recording all the CAN
 > [!IMPORTANT]
 > This project does not use the standard EVT-core abstraction layer and corresponding built tools! 
 
-Before building this project, the ESP-IDF tool chains needs to be installed. Follow the [install guide](https://github.com/espressif/vscode-esp-idf-extension/blob/master/docs/tutorial/install.md) from Espressif. 
+Before building this project, the ESP-IDF tool chains needs to be installed. There is an [install guide](https://github.com/espressif/vscode-esp-idf-extension/blob/master/docs/tutorial/install.md) provided by Espressif that should be referenced if any issue arise. First, install Visual Studio Code and the "ESP-IDF" extension. Once installed click on the "ESP-IDF: Explorer" icon on the left side bar. Then the "Configure ESP-IDF Extension" option under the commands. In this menu, click on "Express" and under the ESP-IDF version, select v5.1.1 (release). Once installed, the rest of the project can be built and flashed to the board. 
+
+> If you encounter `esp_lcd_panel_rgb.c:747:1: internal compiler error: Segmentation fault`, pray it goes away.
 
 Once the build tools are setup, clone this repository with:
 ```
 git clone --recurse-submodules https://github.com/RIT-EVT/GUB2.git
 ```
 Once the repository is cloned, the project can be built in VSCode using the ESP-IDF extension and pressing the build button on the bottom status bar. The project can also be built with `cmake build` from the ESP-IDF Terminal.
 
+## Debugging
+For debugging the board you will likely have to launch openocd manually as it will often times fail. To do so open the ESP-IDF terminal and enter one of the two following commands depending on the board. 
+
+For the ESP32-S3-DevKitC-1 board or built in usb debugger: 
+`openocd -f "board/esp32s3-builtin.cfg"` \
+For the GUB:
+`openocd -s share/openocd/scripts -f interface/ftdi/esp32_devkitj_v1.cfg -f board/esp32s3-ftdi.cfg`
 
+<!-- ## Project Structure -->
+
+## Additional notes
+The Freertos tick rate has been set to 1000 Hz (1ms). This is the smallest unit of delay a task can wait for! Current project has about a 1-2sec startup time.
```

Comment:
Hard to say why it takes so long. Part of it may be the boot loader. Also it wasn't a super scientific test that consited of hand timing from the release of the reset button to the heartbeat LED turning on.

Reply:
Well, that's probably not super accurate. We should investigate this more when we have the full system set up to test with